### ISSUE TITLE
can't modify content type; autoReload is not available

### ISSUE THREAD

**Issue by JamMaster1999** [strapi/strapi#8729]
### Describe the bug
The content type builder does not allow me to modify any fields. 
I get the error autoReload must be enabled for this plugin(content type builder)

### Steps to reproduce the behavior

1. Go to terminal
2. run `npm run develop`
![image](https://user-images.githubusercontent.com/46285145/100120009-55ed8400-2e2c-11eb-8f6c-5ba2817df29b.png)

### Expected behavior

When starting server in the development, I should have access to content type builder. 

### Code snippets

server.js config in the development:
`module.exports = ({ }) => ({ "host": "localhost", "port": 1337, "autoReload": { "enabled": true } }); `
server.js config in the production:
`module.exports = ({ env }) => ({ host: env('HOST', '0.0.0.0'), port: env.int('PORT', 1337), url: env('', 'https://glacial-river-01756.herokuapp.com'), admin: { auth: { secret: env('ADMIN_JWT_SECRET') }, }, });`
### System

- Node.js version: 12.18.14
- NPM version:
- Strapi version: 3.3.3
- Database: MongoDB
- Operating system: Mac OS X Big sur

**Comment by derrickmehaffy** [comment]
@JamMaster1999 I'm unable to reproduce your issue, we will need more steps to reproduce.

![image](https://user-images.githubusercontent.com/8593673/100123811-c21db700-2e2f-11eb-89eb-90c3df7738b7.png)

**Comment by JamMaster1999** [comment]
What are the steps/files I could provide? 
Maybe I could provide with source files?

database.js is as follows: `module.exports = ({ env }) => ({
  defaultConnection: 'default',
  connections: {
    default: {
      connector: 'mongoose',
      settings: {
        uri: env('DATABASE_URI'),
      },
      options: {
        ssl: true,
      },
    },
  },
});`

**Comment by derrickmehaffy** [comment]
The database looks fine (and shouldn't interfere with the content-type builder or CTB).

Can you please try doing a clean build using `npm run build -- --clean` (the extra double dash isn't a typo, npm scripts require it unless you are using yarn which allows for `yarn build --clean`) with no `NODE_ENV`

**Comment by JamMaster1999** [comment]
I was able to figure it out. In server.js for both environments(production and development) I needed to have admin_jwt. 

Thanks for your help @derrickmehaffy

**Comment by derrickmehaffy** [comment]
Ah that's quite odd, I've never seen that kill the CTB before, it usually just throws an error and won't let you login.

**Comment by amojarr** [comment]
Hi @JamMaster1999 
I am having the same issue. Could you please provide me the details of what should be in the server.js? 
This is what i have inside config/server.js currently. 
module.exports = ({ env }) => ({
  host: env('HOST', '0.0.0.0'),
  port: env.int('PORT', 1337),
  admin: {
    auth: {
      secret: env('ADMIN_JWT_SECRET', 'SOME_SECRET),
    },
  },
});

Thank you in advance.

**Comment by chrisegb** [comment]
Hi, @derrickmehaffy @JamMaster1999  I am facing the same issue, the difference here is that:

1. I already have my NODE_ENV variable with development value
2. I put the value "127.0.0.1" into the HOST in the server.js file into the config/ folder
3. I also have the secret: env('ADMIN_JWT_SECRET', 'cc7ca76e799c0185736c87e762d0c8f5') into the server.js file into the config/ fodler
4. I ran npm run build -- --clean
5. I ran npm run strapi develop --watch-admin
6. I ran echo $NODE_ENV and the output is "development" (I set that manually with an export).

I have been searching in internet about the solution for this and I still unable to see the "Create new Collection Type" or even modify an existing one.

I am using:
Strapi 3.3.2 Community Edition
Node v12.20.1
NPM 6.14.10

I don't know I am still unable to see the "create new collection type" or even edit an existing one.

**Comment by chrisegb** [comment]
![Screenshot from 2021-04-26 12-11-18](https://user-images.githubusercontent.com/47096577/116123448-9d509880-a688-11eb-908f-0b0f43cdca53.png)
![Screenshot from 2021-04-26 12-11-36](https://user-images.githubusercontent.com/47096577/116123450-9de92f00-a688-11eb-8344-4956d097ae3f.png)

**Comment by derrickmehaffy** [comment]
The issue is `I ran npm run strapi develop --watch-admin`, at the moment I think the autoReload is disabled while using the --watch-admin as that flag is used to reload the admin panel (for modifying the admin).

I would suggest building the models first before running the --watch-admin.

**Comment by chrisegb** [comment]
@derrickmehaffy but the point here is: Why the "Create new Collection Type" button does not appear?

**Comment by derrickmehaffy** [comment]
If the autoReload boolean internally is not set to true, the admin will not show those buttons.

**Comment by chrisegb** [comment]
Sorry @derrickmehaffy in this case this is what I have in my server.js
```
module.exports = ({ env }) => ({
  host: env('HOST', 'localhost'),
  port: env.int('PORT', 1337),
  url: 'https://strapi.spideraf.com',
  admin: {
    auth: {
      secret: env('ADMIN_JWT_SECRET', 'cc7ca76e799c0185736c87e762d0c8f5')
    }
  },
  autoReload: {
    enabled: true
  }
})
```
Is this enough or is any documentation talking about how to put true for the autoReload feature, strapi doc is very confusing,

**Comment by chrisegb** [comment]
I have also read this: https://strapi.io/documentation/developer-docs/latest/developer-resources/cli/CLI.html#strapi-develop and I ran the strapi develop, but the autoReload seems to be in false but actually I don't have that flag in false in any part. @derrickmehaffy

**Comment by amojarr** [comment]
Are you running strapi via` npm run develop`? That way i see the create content type section.

**Comment by chrisegb** [comment]
@amojarr yes, I am running `npm run develop`. I can see the content type plugin in the left menu but what I can't see is the "+Add new Collection Type" button in the list of collection types once inside the content type plugin.

**Comment by derrickmehaffy** [comment]
> I have also read this: https://strapi.io/documentation/developer-docs/latest/developer-resources/cli/CLI.html#strapi-develop and I ran the strapi develop, but the autoReload seems to be in false but actually I don't have that flag in false in any part. @derrickmehaffy



> Sorry @derrickmehaffy in this case this is what I have in my server.js
> 
> ```
> module.exports = ({ env }) => ({
>   host: env('HOST', 'localhost'),
>   port: env.int('PORT', 1337),
>   url: 'https://strapi.spideraf.com',
>   admin: {
>     auth: {
>       secret: env('ADMIN_JWT_SECRET', 'cc7ca76e799c0185736c87e762d0c8f5')
>     }
>   },
>   autoReload: {
>     enabled: true
>   }
> })
> ```
> 
> Is this enough or is any documentation talking about how to put true for the autoReload feature, strapi doc is very confusing,

We don't check for that variable in the watch-admin command: 

https://github.com/strapi/strapi/blob/5844b9c6b2131ce1b1c7289e175c7557990dacff/packages/strapi/lib/commands/watchAdmin.js#L23-L34

**Comment by chrisegb** [comment]
@derrickmehaffy ok, then how do I put the autoReload flag in true, I am not running the watch-admin command I am just running the npm run develop and what the documentations says is that command is enough to start strapi in development environment with the autoReload flag enabled, but I am unable to see the "Create new Collection Type" button or even modify one.

**Comment by derrickmehaffy** [comment]
Try to rebuild your admin as you may have accidentally built the admin with a different environment.

`NODE_ENV=development npm run build -- --clean` (the NODE_ENV isn't required, I'm only including it to clarify the admin has to be built with the development environment, if you don't have that var then development is the default)

**Comment by chrisegb** [comment]
As you can see in the images below I have built the strapi in development environment, do you have any other advise?
(Also I checked and I don't have nothing assign to the NODE_ENV variable)
![Screenshot from 2021-04-29 19-19-46](https://user-images.githubusercontent.com/47096577/116633905-4b29a480-a920-11eb-86aa-2068e0443266.png)
![Screenshot from 2021-04-29 19-20-05](https://user-images.githubusercontent.com/47096577/116633906-4bc23b00-a920-11eb-865d-49692cfcc023.png)
![Screenshot from 2021-04-29 19-20-32](https://user-images.githubusercontent.com/47096577/116633907-4bc23b00-a920-11eb-9b3c-2d1c1901f302.png)
![Screenshot from 2021-04-29 19-22-05](https://user-images.githubusercontent.com/47096577/116633908-4c5ad180-a920-11eb-9d96-e020dce7e4bd.png)

**Comment by am2222** [comment]
I have the same issue and tried all of the above methods and could not find any solutions yet. 
https://github.com/strapi/strapi/issues/5631

**Comment by chrisegb** [comment]
I got the opportunity to change from projects but I found the solution, check the server.js file I think, there you will find the default I of the host, change that to localhost, that was my solution, if you want we can check it together since I don't have the project anymore and I poorly remember the folder structure.

**Comment by am2222** [comment]
@chrisegb Thanks, Yes I also found the solution. we should comment `url` parameter from `server.js` file otherwise it keeps disabling `autoReload`

**Comment by cgmiele** [comment]
i had this issue and what fixed it for me was changing my NODE_ENV variable from "dev" to "development". hope this can help someone.

**Comment by taviroquai** [comment]
@am2222 thank you, removing url from the server.js config and build again solved for me.
Is this documented anywhere? I was going crazy, this is not so obvious.

**Comment by TofanIulian** [comment]
For me the issue happens randomly. Some days I would start the project locally and the Content-Type Builder would complain that the autoReload feature needs to be enabled and wouldn't display the button to add new types, while on other days it would work just fine.

Sometimes a restart of vscode/my computer would do the trick.

Another thing I just discovered is that if I ran `npm run develop` instead of `npm run start` the problem seems to resolve itself. Don't know if this fixes the problem permanently but it sure did today.

Hope i was of any help.

Strapi version: 4.5.1
`
  "scripts": {
    "develop": "strapi develop",
    "start": "strapi start",
  },
`
